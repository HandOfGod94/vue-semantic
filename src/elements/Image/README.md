# Image

An image is a graphic representation of something

- [Image](#image)
    - [SeImage](#seimage)
        - [Name](#name)
        - [Props](#props)

> Note: All * makred are required attributes

## SeImage
### Name 
`se-image`
### Props
| Name           | Default | Type    | Description                                                                                 |
| -------------- | ------- | ------- | ------------------------------------------------------------------------------------------- |
| as             | `img`   | String  | An element type to render as (String)                                                       |
| src*           |         | String  | Soruce url for image **(for Vue it comes from data object)**                                |
| size           |         | String  | An image can have different sizes(mini,tiny,huge,massive)                                   |
| hidden         | `false` | Boolean | An image can be hidden                                                                      |
| disabled       | `false` | Boolean | An image can be disabled                                                                    |
| avatar         | `false` | Boolean | An image may be formatted to appear inline with text as an avatar                           |
| bordered       | `false` | Boolean | An image may include a border to emphasize the edges of white or transparent content        |
| fluid          | `false` | Boolean | An image can take up the width of its container                                             |
| rounded        | `false` | Boolean | An image may appear rounded                                                                 |
| circular       | `false` | Boolean | An image may appear circular                                                                |
| vertical-align | `false` | Boolean | An image can specify its vertical alignment                                                 |
| centered       | `false` | Boolean | An image can appear centered in a content block                                             |
| spaced         | `false` | Boolean | An image can specify that it needs an additional spacing to separate it from nearby content |
| float          |         | String  | An image can sit to the left or right of other content                                      |
