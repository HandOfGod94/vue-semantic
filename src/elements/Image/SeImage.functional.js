export default {
  functional: true,
  render (h, ctx) {
    const attrs = ctx.data.attrs
    const Component = attrs.as

    return (
      <Component class={ctx.data.class} src={attrs.src}>
        {attrs.slots.default}
      </Component>
    )
  }
}
