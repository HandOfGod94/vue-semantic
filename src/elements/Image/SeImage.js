import SeImageComponent from './SeImage.functional'

export default {
  name: 'se-image',
  props: {
    as: { type: String, default: 'img' },
    src: { type: String, require: true },
    size: String,
    hidden: Boolean,
    disabled: Boolean,
    avatar: Boolean,
    bordered: Boolean,
    fluid: Boolean,
    rounded: Boolean,
    circular: Boolean,
    verticalAlign: String,
    centered: Boolean,
    spaced: [Boolean, String],
    float: String
  },

  computed: {
    classObject () {
      return {
        ui: true,
        hidden: this.hidden,
        disabled: this.disabled,
        [this.size]: this.size,
        bordered: this.bordered,
        avatar: this.avatar,
        fluid: this.fluid,
        rounded: this.rounded,
        circular: this.circular,
        [`${this.verticalAlign} aligned`]: this.verticalAlign,
        centered: this.centered,
        [`${this.spaced} spaced`]: !!this.spaced,
        [`${this.float} floated`]: this.float,
        image: true
      }
    }
  },

  render (createElement) {
    return createElement(SeImageComponent, {
      attrs: {
        ...this.$props,
        slots: this.$slots
      },
      class: this.classObject
    })
  }
}
