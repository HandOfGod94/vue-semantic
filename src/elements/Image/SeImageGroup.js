import Vue from 'vue'

export default {
  name: 'se-image-group',
  props: {
    size: String
  },
  computed: {
    classes: function () {
      let classes = {
        ui: true,
        [this.size]: this.size,
        images: true
      }
      return classes
    }
  },
  render: function (h) {
    return (
      <div class={this.classes}>
        {this.$slots.default}
      </div>
    )
  }
}
