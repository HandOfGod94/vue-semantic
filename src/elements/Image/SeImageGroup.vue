<template>
  <div :class="classObject">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'se-image-group',
  props: {
    size: String
  },
  computed: {
    classObject () {
      return {
        ui: true,
        [this.size]: this.size,
        images: true
      }
    }
  }
}
</script>
