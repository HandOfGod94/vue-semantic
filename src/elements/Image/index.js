export { default as SeImage } from './SeImage'
