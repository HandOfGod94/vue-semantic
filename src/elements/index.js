import SeButton from './Button/SeButton'
import SeButtonGroup from './Button/SeButtonGroup'
import SeContainer from './Container/SeContainer'
import SeDivider from './Divider/SeDivider'
import SeHeader from './Header/SeHeader'
import SeSubHeader from './Header/SeSubHeader'
import SeImage from './Image/SeImage'
import SeImageGroup from './Image/SeImageGroup'
import SeLabel from './Label/SeLabel'
import SeLabelDetail from './Label/SeLabelDetail'
import SeLabelGroup from './Label/SeLabelGroup'
import SeInput from './Input/SeInput'

export {
  SeButton,
  SeButtonGroup,
  SeContainer,
  SeDivider,
  SeHeader,
  SeSubHeader,
  SeImage,
  SeImageGroup,
  SeLabel,
  SeLabelDetail,
  SeLabelGroup,
  SeInput
}
