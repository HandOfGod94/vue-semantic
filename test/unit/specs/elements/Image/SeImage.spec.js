import { SeImage } from '@/elements/Image'
import { mount } from '@vue/test-utils'

const imageContent = '<img src="static/user.png">'

describe('SeImage snapshot test', () => {
  it('should match with exisiting snapshot', () => {
    const wrapper = mount(SeImage)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SeImage classes test', () => {
  it('should have size class', () => {
    const data = { propsData: { size: 'small' } }
    const wrapper = mount(SeImage, data)
    expect(wrapper.classes()).toContain('small')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have img as child', () => {
    const data = { propsData: { as: 'a' }, slots: { default: imageContent } }
    const wrapper = mount(SeImage, data)
    expect(wrapper.contains('a.ui.image>img')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have disabled class', () => {
    const data = { propsData: { disabled: true } }
    const wrapper = mount(SeImage, data)
    expect(wrapper.classes()).toContain('disabled')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have avatar class', () => {
    const data = { propsData: { avatar: true } }
    const wrapper = mount(SeImage, data)
    expect(wrapper.classes()).toContain('avatar')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have bordered class', () => {
    const data = { propsData: { bordered: true } }
    const wrapper = mount(SeImage, data)
    expect(wrapper.classes()).toContain('bordered')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have bordered class', () => {
    const data = { propsData: { verticalAlign: 'middle' } }
    const wrapper = mount(SeImage, data)
    expect(wrapper.contains('img.ui.middle.aligned.image')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have centered class', () => {
    const data = { propsData: { verticalAlign: 'centered' } }
    const wrapper = mount(SeImage, data)
    expect(wrapper.classes()).toContain('centered')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
