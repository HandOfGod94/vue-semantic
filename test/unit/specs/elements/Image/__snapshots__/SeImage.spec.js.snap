// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeImage classes test should have avatar class 1`] = `<img class="ui avatar image">`;

exports[`SeImage classes test should have bordered class 1`] = `<img class="ui bordered image">`;

exports[`SeImage classes test should have bordered class 2`] = `<img class="ui middle aligned image">`;

exports[`SeImage classes test should have centered class 1`] = `<img class="ui centered aligned image">`;

exports[`SeImage classes test should have disabled class 1`] = `<img class="ui disabled image">`;

exports[`SeImage classes test should have img as child 1`] = `
<a class="ui image">
    <img src="static/user.png">
</a>
`;

exports[`SeImage classes test should have size class 1`] = `<img class="ui small image">`;

exports[`SeImage snapshot test should match with exisiting snapshot 1`] = `<img class="ui image">`;
